### ISSUE TITLE
Unique ID

### ISSUE THREAD

**Issue by ronaldt80** [GeorgeSG/philips_airpurifier_http#5]
George, any particular reason why you removed the unique id from this component? If not, mind if I launch a PR incorporating this once more?

**Comment by GeorgeSG** [comment]
Yeah, I removed it because it was not working consistently. It depended on acquiring the Mac address of the device via the API. I don't know if it will even work over COAP.

People used to have issues adding multiple devices, which was also due to the unique_id implementation. This may have been a bug, but it was hard for me to debug it (no 2nd device), and I know the current approach w/o unique_id works.

If you want to change the entity id, you can set the `name` property. 


If you can implement it in a stable way and verify that it works, I'm not against reintroducing unique_id.

**Comment by ronaldt80** [comment]
I raised the issue initially on the original implementation. Your former
development branch however solved my issues and works with 2 machines. I
have had it run consistently for almost a year. Does indeed not work with
coap though but betaboon has found a different solution, will try to
reverse engineer this. Otherwise perhaps use a random number generator
instead? Or link it to the last digits of the IP address?

**Comment by GeorgeSG** [comment]
I'm not sure about using random numbers. I think the guidelines are to use something tied to the device, and if the device has an IP, use the mac address. 

There's a good discussion [here](https://github.com/home-assistant/architecture/issues/333).

Also, I see that betaboon's component uses their own custom python package. Please try implementing this via `py-air-control`, I'd like to stick to that if possible.

**Comment by GeorgeSG** [comment]
Also, I'm planning to convert this to config flow to be future proof.

**Comment by ronaldt80** [comment]
> Also, I see that betaboon's component uses their own custom python package. Please try implementing this via `py-air-control`, I'd like to stick to that if possible.

I tried the py-aircrtl implementation of COAP; it is flawed and does not work well with the newer machines. The purifier crashes as a consequence of the continuous polling by HA. 

I made it working based on your component, using the COAP client instead of HTTP; behavior was erratic - device was turning unavailable after 20 seconds and reappearing after a minute or so. The pyairctrl output (CLI) is not much better.



There are quite some discussions on the HA community forum on this.

**Comment by d4g** [comment]
Is there any update on this?

**Comment by ronaldt80** [comment]
> Is there any update on this?

you can check my fork - I implemented the unique id based on Mac Address. Works fine for me. It's a requirement to have a unique ID to expose the purifier to voice assistants.

BTW - don't use my complete fork unless you have the AC3259; I hard coded the sleep mode specifically for the API of this model.

**Comment by d4g** [comment]
@ronaldt80 could you please raise a pull request for the unique id? (Only)

**Comment by ronaldt80** [comment]
Yes, will do

**Comment by jonnytinkererofthings** [comment]
Just came here to raise about this very thing, as the lack of unique ID is stopping an Area to be applied, unless I'm missing something. With areas being used more and more, especially with Assist and Voice PE, it would be great to be able to set a unique_id so that an Area can be set. 

(apologies if this has been raised in the wrong place, let me know if I should raise a whole new issue for Areas)
